### DISCUSSION TITLE
Report template without updating containers?

### DISCUSSION THREAD

**Discussion by m3Lith** [containrrr/watchtower#1593] [Q&A] [answered]
I want to use Watchtower to notify me via Discord about image updates, without actually updating the containers.

I use `WATCHTOWER_MONITOR_ONLY` for this.
By default, I get the expected notification (4 outdated containers):

![image](https://user-images.githubusercontent.com/1583788/225376001-69d0f7e6-7a35-4c4d-ac2e-f6364c8c012c.png)

However, if I enable `WATCHTOWER_NOTIFICATION_REPORT` and use the  "custom report template that always sends a session report after each run" (from https://containrrr.dev/watchtower/notifications/), I get the opposite result (a list of up-to-date containers):

![image](https://user-images.githubusercontent.com/1583788/225376660-4d0bb793-fe0e-46a6-a822-b016cd58c50d.png)

In both setups, the logs show the same:

![image](https://user-images.githubusercontent.com/1583788/225377104-6d1406ce-27ed-45a2-91dd-63b4f6d45d54.png)

So my question. Is it possible to use the report template together with `WATCHTOWER_MONITOR_ONLY` to see outdated images?

**Comment by m3Lith** [comment, marked as answer]
Update: I guess I found what I needed here in #1469. However, is there a list of all the variables that can be used in the template?

**Reply by piksel** [reply]
Yeah, we really should include a "map" of the template data model in the docs, but at least, here it for reference:
https://github.com/containrrr/watchtower/blob/9470bf81c5c9bfee4e553a3562a82a86d4f31cc3/pkg/notifications/model.go#L8-L19

and the `.Report` is a set of container reports grouped by state:
https://github.com/containrrr/watchtower/blob/9470bf81c5c9bfee4e553a3562a82a86d4f31cc3/pkg/types/report.go#L4-L12

...which have the following fields:
https://github.com/containrrr/watchtower/blob/9470bf81c5c9bfee4e553a3562a82a86d4f31cc3/pkg/types/report.go#L15-L23
